src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/IListPackageReportWriter.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Configuration;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal interface IListPackageReportWriter
```

Comment:
```suggestion
    internal interface IListPackageConsoleReportWriter
```

The methods in this interface are too specific to the console output to be re-usable when JSON output will be implemented.

I guess it would be quite a lot of effort, hopefully I'm wrong, but a more MVC approach of

```cs
interface IListPackageReportWriter
{
   void Write(ProjectInfo projectInfo);
}
```

would make me a lot happier.

For example, this interface, with different APIs to write headers, write updates, deprecation, vulnerability, etc. The caller needs to call these APIs in a certain order. But if a different report type wants to write different sections in a different order, it becomes very difficult.

If we rename the interface to be specific to the console writer, then tests written for it will still be relevant when JSON output is implemented and the code is refactored to support the very different output requirements. If that work can be front-loaded now, to make `ListPackage` MVC-like, then that would be great. But I understand if that's too much work, in which case I don't think we should pretend that this interface is usable for anything other than console output.